cmd/installer/pkg/install/install.go

```diff
@@ -103,7 +105,7 @@ func NewInstaller(cmdline *procfs.Cmdline, seq runtime.Sequence, opts *Options)
 	bootLoaderPresent := i.bootloader != nil
 
 	if !bootLoaderPresent {
-		i.bootloader, err = bootloader.New()
+		i.bootloader, err = bootloader.New(i.options.UEFI)
```

Comment:
we should probably rename `bootloader.New()` to `bootloader.Detect()` or something like that, and call it only when running as `installer`, while wile running as `imager` we should never call `.Detect`, and instead use explicitly either `grub.New()` or `sdboot.New()` depending on the image type.

otherwise image type depends on the environment used to run the `imager`

Reply:
this might be a good way to replace hacks which detect `/dev/loop` and assume image creation with a proper explicit flag  - which mode we run right now:

* image creation
* install/upgrade on the node